test/common/include/alpaka/test/queue/QueueCpuOmp2Collective.hpp

```diff
@@ -0,0 +1,454 @@
+                    {
+                        // wait for all tasks en-queued before the parallel region
+                        while(!queue::empty(*queue.m_spBlockingQueue)){}
+                        #pragma omp barrier
+                    }
+                    else
+                    {
+                        std::lock_guard<std::mutex> lk(queue.m_spQueueImpl->m_mutex);
+                        wait::wait(*queue.m_spBlockingQueue);
+                    }
+                }
+            };
+
+
+            //#############################################################################
+            //! The CPU OpenMP2 collective device queue event wait trait specialization.
+            template<>
+            struct WaiterWaitFor<
+                std::shared_ptr<queue::cpu::detail::QueueCpuOmp2CollectiveImpl>,
+                event::EventCpu>
+            {
+                //-----------------------------------------------------------------------------
+                ALPAKA_FN_HOST static auto waiterWaitFor(
+                    std::shared_ptr<queue::cpu::detail::QueueCpuOmp2CollectiveImpl> &,
+                    event::EventCpu const &)
+                -> void
+                {
+                    #pragma omp barrier
+                }
+            };
+            //#############################################################################
+            //! The CPU OpenMP2 collective queue event wait trait specialization.
+            template<>
+            struct WaiterWaitFor<
+                queue::QueueCpuOmp2Collective,
+                event::EventCpu>
+            {
+                //-----------------------------------------------------------------------------
+                ALPAKA_FN_HOST static auto waiterWaitFor(
+                    queue::QueueCpuOmp2Collective & queue,
+                    event::EventCpu const & event)
+                -> void
+                {
+                    if(::omp_in_parallel() != 0)
+                    {
+                        // wait for all tasks en-queued before the parallel region
+                        while(!queue::empty(*queue.m_spBlockingQueue)){}
+                        wait::wait(queue);
+                    }
+                    else
+                        wait::wait(*queue.m_spBlockingQueue, event);
+                }
+            };
+        }
+    }
+    //-----------------------------------------------------------------------------
+    //! The test specifics.
+    namespace test
+    {
+        //-----------------------------------------------------------------------------
+        //! The test queue specifics.
+        namespace queue
+        {
+            namespace traits
+            {
+                //#############################################################################
+                //! The blocking queue trait specialization for a OpenMP2 collective CPU queue.
+                template<>
+                struct IsBlockingQueue<
```

Comment:
Could we move the `IsBlockingQueue` trait for this and all the other queues to the main `alpaka` library?
Could we please rename this queue to `QueueCpuBlockingOmp2Collective` so hat it matches the style of all others?

Reply:
> Could we please rename this queue to `QueueCpuBlockingOmp2Collective` so that it matches the style of all others?

I also thought at this name before but it is not blocking within a parallel region so I avoided to use blocking within the name

Reply:
> Could we move the `IsBlockingQueue` trait (maybe as `IsBlocking`) for this and all the other queues to the main `alpaka` library?

We can do it but outside of this PR.

Reply:
But why does `IsBlockingQueue` then return true? A queue that is blocking is defined by the fact that a call to `queue::enqueue(queue, task);` (which is also hidden within `kernel::exec`) does not return before the task is executed. Is this true or false for this queue?

Reply:
because it depends where you use it (in a parallel region or not) outside it is blocking and to not move over tests those require a nonblocking queue it is set to blocking.